x/oracle/keeper/keeper_test.go

```diff
@@ -0,0 +1,451 @@
+				shouldPass: false,
+				contains:   "collateral not enough",
+			},
+		},
+	}
+	for _, tc := range tests {
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			err := suite.keeper.CreateOperator(suite.ctx, tc.args.senderAddr, sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.collateral)}, tc.args.proposerAddr, tc.args.operatorName)
+			suite.Require().NoError(err, tc.name)
+			collateral, err := suite.keeper.GetCollateralAmount(suite.ctx, tc.args.senderAddr)
+			suite.Require().NoError(err, tc.name)
+			suite.Equal(sdk.NewInt(tc.args.collateral), collateral)
+			// add collateral
+			suite.NoError(suite.keeper.AddCollateral(suite.ctx, tc.args.senderAddr, sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.collateralToAdd)}))
+			// check collateral amount
+			collateral, err = suite.keeper.GetCollateralAmount(suite.ctx, tc.args.senderAddr)
+			if tc.errArgs.shouldPass {
+				if tc.args.reduce {
+					// reduce collateral
+					suite.NoError(suite.keeper.ReduceCollateral(suite.ctx, tc.args.senderAddr, sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.collateralToReduce)}))
+					collateral, err = suite.keeper.GetCollateralAmount(suite.ctx, tc.args.senderAddr)
+					suite.Require().NoError(err, tc.name)
+					suite.Equal(sdk.NewInt(tc.args.collateral+tc.args.collateralToAdd-tc.args.collateralToReduce), collateral)
+				} else {
+					suite.Require().NoError(err, tc.name)
+					suite.Equal(sdk.NewInt(tc.args.collateral+tc.args.collateralToAdd), collateral)
+				}
+			} else {
+				if tc.args.reduce {
+					// reduce collateral
+					err := suite.keeper.ReduceCollateral(suite.ctx, tc.args.senderAddr, sdk.Coins{sdk.NewInt64Coin("uctk", tc.args.collateralToReduce)})
+					suite.Require().Error(err, tc.name)
+					suite.Require().True(strings.Contains(err.Error(), tc.errArgs.contains))
+				} else {
+					suite.Require().Error(err, tc.name)
+					suite.Require().True(strings.Contains(err.Error(), tc.errArgs.contains))
+				}
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestOperator_Reward() {
+	type args struct {
+		collateral   int64
+		senderAddr   sdk.AccAddress
+		proposerAddr sdk.AccAddress
+		operatorName string
+		rewardToAdd  int64
+		withdrawAll  bool
+	}
+
+	type errArgs struct {
+		shouldPass bool
+		contains   string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		errArgs errArgs
+	}{
+		{"Operator(1) Add: 100000 uctk",
+			args{
+				collateral:   50000,
+				senderAddr:   suite.address[0],
+				proposerAddr: suite.address[1],
+				operatorName: "Operator",
+				rewardToAdd:  100000,
+			},
+			errArgs{
+				shouldPass: true,
+				contains:   "",
+			},
+		},
+		{"Operator(1) WithdrawAll: Reward <= Collateral",
+			args{
+				collateral:   50000,
+				senderAddr:   suite.address[0],
+				proposerAddr: suite.address[1],
+				operatorName: "Operator",
+				rewardToAdd:  50000,
+				withdrawAll:  true,
+			},
+			errArgs{
+				shouldPass: true,
+				contains:   "",
+			},
+		},
+		{"Operator(1) WithdrawAll: Reward > Collateral",
```

Comment:
hmm interesting. Have we found the cause of this yet?

Reply:
Not that I know of